File: src/order-processor-service/src/main/java/com/gabrielaraujo/order_processor_service/adapters/dao/repositories/OrderRepository.java
package com.gabrielaraujo.order_processor_service.adapters.dao.repositories;

import com.gabrielaraujo.order_processor_service.adapters.dao.tables.OrderTable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OrderRepository extends JpaRepository<OrderTable, Integer> {
}

File: src/order-analytics-api/OrderAnalytics.Adapters/DAO/Repositories/OrderRepository.cs
using System.Numerics;
using OrderAnalytics.Adapters.DAO.Context;
using OrderAnalytics.Adapters.DAO.Tables;

namespace OrderAnalytics.Adapters.DAO.Repositories;

public abstract class OrderRepository
{
    protected DatabaseContext _context { get; init; }
    
    public abstract Task<List<OrderTable>> GetOrdersFromClientAsync(int clientId);
    public abstract Task<decimal> GetFinalPriceFromOrderAsync(int orderId);
    public abstract Task<BigInteger> CountOrdersFromClientAsync(int clientId);
}